src/codebook/settings/layout.js

```diff
@@ -0,0 +1,77 @@
+export function layout(codebook) {
+  //Create list of columns in the data file.
+  const columns = Object.keys(codebook.data.raw[0]),
+    groupColumns = codebook.config.groups.map(d => d.value_col),
+    filterColumns = codebook.config.groups.map(d => d.value_col),
+    columnTableColumns = ["Column", "Group", "Filter"], //, 'Visibility', 'Label'],
+    columnMetadata = columns.map(column => {
+      const columnDatum = {
+        Column: column,
+        Group: {
+          type: "checkbox",
+          checked: groupColumns.indexOf(column) > -1
+        },
+        Filter: {
+          type: "checkbox",
+          checked: filterColumns.indexOf(column) > -1
+        }
+        /*,'Visibility':
+                            {type: 'checkbox'
+                            ,checked: true}
+                        ,'Label':
+                            {type: 'text'
+                            ,checked: filterColumns.indexOf(column) > -1}*/
+      };
+
+      return columnDatum;
+    }),
+    columnTable = codebook.settings.wrap
+      .append("table")
+      .classed("column-table", true),
+    columnTableHeader = columnTable.append("thead").append("tr"),
+    columnTableHeaders = columnTableHeader
+      .selectAll("th")
+      .data(columnTableColumns)
+      .enter()
+      .append("th")
+      .attr("class", d => d)
+      .text(d => d),
+    columnTableRows = columnTable
+      .append("tbody")
+      .selectAll("tr")
+      .data(columnMetadata)
+      .enter()
+      .append("tr");
+  columnTableRows.each(function(rowDatum) {
```

Comment:
Any reason to use .each() and a for loop here instead of a more traditional: 

```
rowCells  = columnTableTows.selectAll("td").
data(function(d){
   return { 
       /*build your data object here*/
    })
    .enter()
    .append("td")
```